src/api/procedures/__tests__/setVenueFiltering.ts

```diff
@@ -0,0 +1,139 @@
+
+    when(venueFilteringMock)
+      .calledWith(enabledTicker)
+      .mockResolvedValue(dsMockUtils.createMockBool(true));
+    when(venueFilteringMock)
+      .calledWith(disabledTicker)
+      .mockResolvedValue(dsMockUtils.createMockBool(false));
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    procedureMockUtils.reset();
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    procedureMockUtils.cleanup();
+    dsMockUtils.cleanup();
+  });
+
+  describe('prepareVenueFiltering', () => {
+    it('the procedure method should be defined', () => {
+      expect(setVenueFiltering).toBeDefined();
+    });
+
+    it('calling it should return a new procedure', () => {
+      const boundFunc = setVenueFiltering.bind(mockContext);
+
+      expect(boundFunc).not.toThrow();
+      expect(procedureMockUtils.getInstance<Params, void>(mockContext)).toBeDefined();
+    });
+  });
+
+  it('should throw an error if already enabled', () => {
+    const args = {
+      ticker: enabledTicker,
+      enabled: true,
+    };
+
+    const proc = procedureMockUtils.getInstance<Params, void>(mockContext);
+
+    return expect(prepareVenueFiltering.call(proc, args)).rejects.toThrow(
+      'Venue filtering is already enabled'
+    );
+  });
+
+  it('should throw an error if already disabled', () => {
+    const args = {
+      ticker: disabledTicker,
+      enabled: false,
+    };
+
+    const proc = procedureMockUtils.getInstance<Params, void>(mockContext);
+
+    return expect(prepareVenueFiltering.call(proc, args)).rejects.toThrow(
+      'Venue filtering is already disabled'
+    );
+  });
+
+  it('should return a set venue filtering transaction spec', async () => {
+    const proc = procedureMockUtils.getInstance<Params, void>(mockContext);
+    const setEnabled = false;
+    const transaction = dsMockUtils.createTxMock('settlement', 'setVenueFiltering');
+
+    const result = await prepareVenueFiltering.call(proc, {
+      ticker: enabledTicker,
+      enabled: setEnabled,
+    });
+
+    expect(result).toEqual({
+      transaction,
+      args: [enabledTicker, setEnabled],
+      resolver: undefined,
+    });
+  });
+
+  describe('getAuthorization', () => {
+    it('should return the appropriate roles and permissions', () => {
+      const proc = procedureMockUtils.getInstance<Params, void>(mockContext);
+      const boundFunc = getAuthorization.bind(proc);
+      const args: Params = {
+        ticker: enabledTicker,
+        enabled: true,
+      };
+
+      expect(boundFunc(args)).toEqual({
+        permissions: {
+          transactions: [TxTags.settlement.SetVenueFiltering],
+          assets: [expect.objectContaining({ ticker: enabledTicker })],
+        },
+      });
+
+      args.enabled = false;
+
+      expect(boundFunc(args)).toEqual({
+        permissions: {
+          transactions: [TxTags.settlement.SetVenueFiltering],
+          assets: [expect.objectContaining({ ticker: enabledTicker })],
+        },
+      });
```

Comment:
Since the `getAuthorization` method doesn't depend on `enabled` param, you can skip this.